src/main/java/Duke.java

```diff
@@ -1,10 +1,89 @@
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import javafx.scene.layout.Region;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+
+/**
+ * Duke is a chat bot program that builds a to do list. Current functions include:
+ * list, delete, done, todo, deadline, event, bye.
+ *
+ */
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+    /**
+     * Constructor for main class for the program.
+     * @param filePath relative path of the file that the data of to do list is saved in.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+    public Duke() {
+
+    }
+
+    /**
+     * Main driver for Duke program.
+     */
+    public void run() {
+        ui.welcomeMessage();
+        tasks.showCurrentTasks();
+        ui.showLineBreak();
+        boolean isFinished = false;
+        while (!(isFinished)) {
+            try {
+                String inputFromUser = ui.handleInput();
+                Command c = Parser.parse(inputFromUser);
+                if (c == null) {
```

Comment:
I think you should put this `c == null` check in the `Command` class instead. in this `Main`/`Duke` class we should only call the method such as `c.isCommandValid()`.